f5/bigip/resource.py

```diff
@@ -295,6 +305,37 @@ def _handle_requests_params(self, kwargs):
             requests_params.update({'params': params})
         return requests_params
 
+    def _check_exclusive_parameters(self, **kwargs):
+        """Check for mutually exclusive attributes in kwargs.
+
+        :raises ExclusiveAttributesPresent
+        """
+        if len(self._meta_data['exclusive_attributes']) > 0:
+            attr_set = set(kwargs.keys())
+            ex_set = set(self._meta_data['exclusive_attributes'][0])
+            common_set = attr_set.intersection(ex_set)
+            if len(common_set) > 1:
+                cset = ', '.join(common_set)
+                error = 'Mutually exclusive arguments submitted. ' \
+                        'The following arguments cannot be set ' \
+                        'together: "%s".' % cset
+                raise ExclusiveAttributesPresent(error)
+
+    @staticmethod
+    def _check_required_parameters(rqset, **kwargs):
+        """Helper function to do operation on sets
+
+        ::returns bool and variable
+        """
+        key_set = set(kwargs.keys())
+        required_minus_received = rqset - key_set
+        if required_minus_received != set():
```

Comment:
We can do the following:

https://gist.github.com/wojtek0806/5084d2d45bc0036dc188aafa3cae44dd

This will allow us to do something like this when using this helper method:

https://gist.github.com/wojtek0806/10d6af27ad8526c44f9135c3afa3d86b

The problem is we will be losing custom exception names like 'MissingRequiredReadParameter'. We can rectify this by doing the following:

https://gist.github.com/wojtek0806/4255aa0e95f1f83c0bf706c685d5246d

We still loose traceback, there are obviously ways we can get that back but they are not recommended and hacky, and it varies quite between v2 and v3 python. I would just go with the first solution, and just have one generic exception name for missing required attributes, as they point to the same thing, does not matter if it is read, create or command attribute